agenta-cli/sdk.py

```diff
@@ -0,0 +1,652 @@
+from typing import Any, Dict, Optional
+
+from pydantic import BaseModel
+
+
+class Reference(BaseModel):
+    # IDENTIFIER
+    id: Optional[str] = None
+    slug: Optional[str] = None
+    # VERSION
+    version: Optional[str] = None
+
+
+class Application(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
+    # HEADER
+    name: Optional[str] = None
+    description: Optional[str] = None
+
+
+class Service(BaseModel):
+    # INTERNALS
+    url: str
+
+
+class Configuration(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
+    # HEADER
+    name: Optional[str] = None
+    description: Optional[str] = None
+    # VERSION
+    version: str
+    latest: bool
+
+    # REFERENCES
+    application: Application
+
+    # INTERNALS
+    parameters: Dict[str, Any]
+
+
+class Variant(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
+    # HEADER
+    name: Optional[str] = None
+    description: Optional[str] = None
+    # VERSION
+    version: str
+    latest: bool
+
+    # REFERENCES
+    application: Application
+    service: Service
+    configuration: Configuration
+
+
+class Environment(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
+    # HEADER
+    name: Optional[str] = None
+    description: Optional[str] = None
+    # VERSION
+    version: str
+    latest: bool
+
+    # REFERENCES
+    application: Application
+    variant: Variant
+
+
+class Prompt(Variant):
+    pass
+
+
+class ApplicationsClient:
+    def __init__(self):
+        pass
+
+    def create(
+        self,
+        *,
+        application_slug: Optional[str] = None,
+    ) -> Prompt:
+        pass
+
+    def fetch(
+        self,
+        *,
+        configuration_reference: Optional[Reference] = None,
+        variant_reference: Optional[Reference] = None,
+        environment_reference: Optional[str] = None,
+    ) -> Prompt:
+        pass
+
+
+class ConfigurationsClient:
+    def __init__(self):
+        pass
+
+    def create(
+        self,
+        *,
+        application_slug: Optional[str] = None,
+        configuration_slug: Optional[str] = None,
+        configuration_parameters: Optional[Dict[str, Any]] = None,
+        configuration_reference: Optional[Reference] = None,
```

Comment:
why do we need a reference here?

Reply:
what is actually the reference? don't we already have a configuration_slug in the call

Reply:
It's true that this SDK was meant to accommodate _all_ _Applications Structure_ features, which won't be true initially. A `reference` (see _DTO_ at the top of the PR) encapsulates either a "commit" id or a slug+version and uniquely identifies a resource.
In our case, this would be called differently and used differently, for now.
Will get back on this.